src/main/java/ch/admin/bar/siardsuite/Controller.java

```diff
@@ -186,18 +195,46 @@ public void updateArchiveMetaData(String dbName, String description, String owne
 
     public void initializeWorkflow(Workflow workflow, RootStage stage) {
         this.model.clearSiardArchive();
-        this.workflow = workflow;
-        stage.openDialog(View.forWorkflow(workflow));
-    }
-
-    public void initializeExport(RootStage stage) {
-        this.workflow = Workflow.EXPORT;
-        stage.openDialog(View.EXPORT_SELECT_TABLES);
-    }
-
-    public void initializeUpload(RootStage stage) {
-        this.workflow = Workflow.UPLOAD;
-        stage.openDialog(View.UPLOAD_DB_CONNECTION_DIALOG);
+        this.setRecentDatabaseConnection(Optional.empty());
+
+        switch (workflow) {
+            case ARCHIVE:
+                stage.openRecentConnectionsDialogForArchiving(
+                        () -> stage.navigate(View.ARCHIVE_STEPPER),
+                        dbConnection -> {
+                            setRecentDatabaseConnection(Optional.of(dbConnection));
+                            stage.navigate(View.ARCHIVE_STEPPER);
+                        }
+                );
+                break;
+            case OPEN:
+                stage.openSelectSiardFileDialog((file, archive) -> {
+                            setSiardArchive(file.getName(), archive);
+                            stage.navigate(View.OPEN_SIARD_ARCHIVE_PREVIEW);
+                        }
+                );
+                break;
+            case EXPORT:
+                stage.openSelectSiardFileDialog((file, archive) -> {
+                            setSiardArchive(file.getName(), archive);
+                            stage.openDialog(View.EXPORT_SELECT_TABLES);
+                        }
+                );
+                break;
+            case UPLOAD:
+                stage.openSelectSiardFileDialog((file, archive) -> {
+                            setSiardArchive(file.getName(), archive);
+                            stage.openRecentConnectionsDialogForUploading(
+                                    () -> stage.openDialog(View.UPLOAD_STEPPER),
```

Comment:
`openDialog` should be replaced with `navigate` (same on line 231) otherwise canceling the following dialog navigates user to the startview instead of the last step in the upload-stepper.

![2024-01-26_11-58](https://github.com/sfa-siard/siard-suite/assets/24672636/5176e178-d98a-4c0e-baaf-40741935f67f)